hal/mbed_sleep_manager.c

```diff
@@ -14,18 +14,123 @@
  * limitations under the License.
  */
 
+#include "mbed_assert.h"
 #include "mbed_sleep.h"
 #include "mbed_critical.h"
 #include "sleep_api.h"
 #include "mbed_error.h"
 #include <limits.h>
+#include <stdio.h>
 
 #if DEVICE_SLEEP
 
 // deep sleep locking counter. A target is allowed to deep sleep if counter == 0
 static uint16_t deep_sleep_lock = 0U;
 
-void sleep_manager_lock_deep_sleep(void)
+#ifdef MBED_SLEEP_STATS_ENABLED
+
+// Length of the identifier extracted from the driver name to store for logging.
+#define IDENTIFIER_WIDTH 7
+// Number of drivers that can be stored in the structure
+#define STATISTIC_COUNT  10
+
+typedef struct __PACKED sleep_statistic {
```

Comment:
Remove the `__PACKED` it doesn't make any difference here and generally I'm sceptical about use of packed structures, it may have negative effects on performance and put pressure on the bus with unaligned accesses.